applications/asset_tracker_v2/src/modules/location_module.c

```diff
@@ -0,0 +1,630 @@
+			pvt_data = event_data->location.details.gnss.pvt_data;
+			stats.satellites_tracked =
+				event_data->location.details.gnss.satellites_tracked;
+			LOG_DBG("  satellites tracked: %d", stats.satellites_tracked);
+
+			if (event_data->location.datetime.valid) {
+				/* Date and time is in pvt_data that is set above */
+				time_set();
+			}
+			data_send_pvt();
+		}
+		LOG_DBG("  Google maps URL: https://maps.google.com/?q=%.06f,%.06f",
+			event_data->location.latitude, event_data->location.longitude);
+
+		inactive_send();
+		break;
+
+	case LOCATION_EVT_RESULT_UNKNOWN:
+		LOG_DBG("Getting location completed with undefined result");
+		stats.search_time = (uint32_t)(k_uptime_get() - stats.start_uptime);
+		LOG_DBG("  search time: %d", stats.search_time);
+		inactive_send();
+		/* No events such as LOCATION_MODULE_EVT_DATA_READY are sent because
+		 * LOCATION_MODULE_EVT_NEIGHBOR_CELLS_DATA_READY has already been sent earlier
+		 * and hence APP_LOCATION has been set already.
+		 */
+		break;
+
+	case LOCATION_EVT_TIMEOUT:
+		LOG_WRN("Getting location timed out");
+		stats.search_time = (uint32_t)(k_uptime_get() - stats.start_uptime);
+		LOG_DBG("  search time: %d", stats.search_time);
+		if (event_data->method == LOCATION_METHOD_GNSS &&
+		    event_data->error.details.gnss.valid) {
+			stats.satellites_tracked =
+				event_data->error.details.gnss.satellites_tracked;
+			LOG_DBG("  satellites tracked: %d", stats.satellites_tracked);
+		}
+		timeout_send();
+		inactive_send();
+		break;
+
+	case LOCATION_EVT_ERROR:
+		LOG_WRN("Getting location failed");
+		stats.search_time = (uint32_t)(k_uptime_get() - stats.start_uptime);
+		if (event_data->method == LOCATION_METHOD_GNSS &&
+		    event_data->error.details.gnss.valid) {
+			stats.satellites_tracked =
+				event_data->error.details.gnss.satellites_tracked;
+			LOG_DBG("  satellites tracked: %d", stats.satellites_tracked);
+		}
+		SEND_EVENT(location, LOCATION_MODULE_EVT_DATA_NOT_READY);
+		inactive_send();
+		break;
+
+	case LOCATION_EVT_GNSS_ASSISTANCE_REQUEST: {
+		LOG_DBG("Requested A-GPS data");
+#if defined(CONFIG_LOCATION_METHOD_GNSS_AGPS_EXTERNAL)
+		struct location_module_event *location_module_event = new_location_module_event();
+
+		location_module_event->data.agps_request = event_data->agps_request;
+		location_module_event->type = LOCATION_MODULE_EVT_AGPS_NEEDED;
+		APP_EVENT_SUBMIT(location_module_event);
+#endif
+		break;
+	}
+
+	case LOCATION_EVT_GNSS_PREDICTION_REQUEST: {
+		LOG_DBG("Requested P-GPS data");
+#if defined(CONFIG_LOCATION_METHOD_GNSS_PGPS_EXTERNAL)
+		struct location_module_event *location_module_event = new_location_module_event();
+
+		location_module_event->data.pgps_request = event_data->pgps_request;
+		location_module_event->type = LOCATION_MODULE_EVT_PGPS_NEEDED;
+		APP_EVENT_SUBMIT(location_module_event);
+#endif
+		break;
+	}
+
+	case LOCATION_EVT_CELLULAR_EXT_REQUEST:
+		LOG_DBG("Getting cellular request");
+		send_neighbor_cell_update((
```

Comment:
Having the start parantheses on same line as `struct` would read a bit better

Reply:
Oops.